kclvm/compiler_base/error/src/diagnostic/components.rs

```diff
@@ -54,3 +57,350 @@ impl Component<DiagnosticStyle> for Label {
         }
     }
 }
+
+/// `StringWithStyle` is a component of diagnostic to display a string with style.
```

Comment:
s/StringWithStyle/StyledString? 

Besides, using `StyledString<T>` in rustc_errors?

Reply:
fixed